contracts/button/lib.rs

```diff
@@ -497,13 +493,16 @@ pub mod button_game {
 
             self.data.set(&data);
 
-            Self::emit_event(self.env(), Event::GameReset(GameReset { when: now }));
             Ok(())
         }
 
         fn reward_pressiah(&self) -> ButtonResult<()> {
             if let Some(pressiah) = self.data.get().unwrap().last_presser {
                 let reward = self.pressiah_reward();
+                Self::emit_event(
```

Comment:
hmmm, I am not sure how it should be exactly but `mint_reward` which is called later can fail. In such case even though the pressiah was found the reward will not be minted, right? Maybe emit this event if `mint_reward` returns ok variant

Reply:
txs even if chained are atomic - if anything fails, the whole state is reverted, so it will never happen

Reply:
thx for explanantion